controllers/components/kserve/kserve_support.go

```diff
@@ -0,0 +1,268 @@
+	return func(registry feature.FeaturesRegistry) error {
+		servingDeployment := feature.Define("serverless-serving-deployment").
+			Manifests(
+				manifest.Location(Resources.Location).
+					Include(
+						path.Join(Resources.InstallDir),
+					),
+			).
+			WithData(
+				serverless.FeatureData.IngressDomain.Define(&kserve.Spec.Serving).AsAction(),
+				serverless.FeatureData.Serving.Define(&kserve.Spec.Serving).AsAction(),
+				servicemesh.FeatureData.ControlPlane.Define(dsciSpec).AsAction(),
+			).
+			PreConditions(
+				serverless.EnsureServerlessOperatorInstalled,
+				serverless.EnsureServerlessAbsent,
+				servicemesh.EnsureServiceMeshInstalled,
+				feature.CreateNamespaceIfNotExists(serverless.KnativeServingNamespace),
+			).
+			PostConditions(
+				feature.WaitForPodsToBeReady(serverless.KnativeServingNamespace),
+			)
+
+		istioSecretFiltering := feature.Define("serverless-net-istio-secret-filtering").
+			Manifests(
+				manifest.Location(Resources.Location).
+					Include(
+						path.Join(Resources.BaseDir, "serving-net-istio-secret-filtering.patch.tmpl.yaml"),
+					),
+			).
+			WithData(serverless.FeatureData.Serving.Define(&kserve.Spec.Serving).AsAction()).
+			PreConditions(serverless.EnsureServerlessServingDeployed).
+			PostConditions(
+				feature.WaitForPodsToBeReady(serverless.KnativeServingNamespace),
+			)
+
+		servingGateway := feature.Define("serverless-serving-gateways").
+			Manifests(
+				manifest.Location(Resources.Location).
+					Include(
+						path.Join(Resources.GatewaysDir),
+					),
+			).
+			WithData(
+				serverless.FeatureData.IngressDomain.Define(&kserve.Spec.Serving).AsAction(),
+				serverless.FeatureData.CertificateName.Define(&kserve.Spec.Serving).AsAction(),
+				serverless.FeatureData.Serving.Define(&kserve.Spec.Serving).AsAction(),
+				servicemesh.FeatureData.ControlPlane.Define(dsciSpec).AsAction(),
+			).
+			WithResources(serverless.ServingCertificateResource).
+			PreConditions(serverless.EnsureServerlessServingDeployed)
+
+		return registry.Add(
+			servingDeployment,
+			istioSecretFiltering,
+			servingGateway,
+		)
+	}
+}
+
+func defineServiceMeshFeatures(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec) feature.FeaturesProvider {
+	return func(registry feature.FeaturesRegistry) error {
+		authorinoInstalled, err := cluster.SubscriptionExists(ctx, cli, "authorino-operator")
+		if err != nil {
+			return fmt.Errorf("failed to list subscriptions %w", err)
+		}
+
+		if authorinoInstalled {
+			kserveExtAuthzErr := registry.Add(feature.Define("kserve-external-authz").
+				Manifests(
+					manifest.Location(Resources.Location).
+						Include(
+							path.Join(Resources.ServiceMeshDir, "activator-envoyfilter.tmpl.yaml"),
+							path.Join(Resources.ServiceMeshDir, "envoy-oauth-temp-fix.tmpl.yaml"),
+							path.Join(Resources.ServiceMeshDir, "kserve-predictor-authorizationpolicy.tmpl.yaml"),
```

Comment:
I don't know if there's some other unintended consequence, but I've removed one of the entries in this `Include()` section in a secondary commit here, to make it easier to "watch" this AuthorizationPolicy. It appeared to be a modification to this kserve-predictor, so what appeared to be happening was that this would set the resource to one state, and then when the update was processed, it would just change one of the field values. In the file on this line, that that value had a static value, and in the update it was a template parameter, so I just changed this file to have the template parameter.

Reply:
Let see how it goes

Reply:
good catch! we moved around these templates and did not cleanup all the changes